pkg/uki/common.go

```diff
@@ -166,3 +166,47 @@ func copyFile(src, dst string) error {
 
 	return destinationFile.Close()
 }
+
+func AddSystemdConfSortKey(fs v1.FS, artifactDir string, log sdkTypes.KairosLogger) error {
+	return fsutils.WalkDirFs(fs, artifactDir, func(path string, info os.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		// Only do files that are conf files but dont match the loader.conf
+		if !info.IsDir() && filepath.Ext(path) == ".conf" && !strings.Contains(info.Name(), "loader.conf") {
+			log.Logger.Debug().Str("path", path).Msg("Adding sort key to file")
+			conf, err := sdkutils.SystemdBootConfReader(path)
+			if err != nil {
+				log.Errorf("Error reading conf file to extract values %s: %s", conf, path)
+			}
+			// Now check and put the proper sort key
+			var sortKey string
+			// If we have 2 different files that start with active, like with the extra-cmdline, how do we set this?
+			// Ideally if they both have the same sort key, they will be sorted by name so the single one will be first
+			// and the extra-cmdline will be second. This is the best we can do currently without making this a mess
```

Comment:
If the extra was meant to be the "default", it would not be set as extra cmdline. I think extra cmdlines were always meant to be alternative boot entries, not part of the "active", "passive", "recovery" dance.

Reply:
Having said that, if "active" fails to boot, I don't think we would want to try "active_myops" next, would we? The next thing we would want to try would be "passive". Unless I'm missing some user flow.

Reply:
Im not sure, maybe active and active_debug should be tried one after the other?

Thats why this has a big explanation, I guess we can add this then move into a more detailed thing or explain it better?

In any case, this is just the first implementation, usually this would not matter much (the order) as we select explicit entries, so this will come later down the line when people start implementing assessment and such?